openedx/core/djangoapps/user_api/management/commands/retire_user.py

```diff
@@ -34,67 +34,72 @@ class Command(BaseCommand):
-            help='Username to be retired'
-        )
-        parser.add_argument(
-            '--user_email',
-            required=True,
-            type=str,
-            help='User email address.'
+            help='Comma separated file that have username and user_email of the users that needs to be retired'
         )
 
     def handle(self, *args, **options):
         """
         Execute the command.
         """
-
-        username = options['username']
-        user_email = options['user_email']
-        try:
-            user = User.objects.get(username=username, email=user_email)
-        except:
-            error_message = (
-                'Could not find a user with specified username and email '
-                'address. Make sure you have everything correct before '
-                'trying again'
-            )
-            logger.error(error_message)
-            raise CommandError(error_message)  # lint-amnesty, pylint: disable=raise-missing-from
-
-        user_model = get_user_model()
-
         try:
-            with transaction.atomic():
-                # Add user to retirement queue.
-                UserRetirementStatus.create_retirement(user)
-                # Unlink LMS social auth accounts
-                UserSocialAuth.objects.filter(user_id=user.id).delete()
-                # Change LMS password & email
-                user.email = get_retired_email_by_email(user.email)
-                user.set_unusable_password()
-                user.save()
-
-                # TODO: Unlink social accounts & change password on each IDA.
-                # Remove the activation keys sent by email to the user for account activation.
-                Registration.objects.filter(user=user).delete()
-
-                # Delete OAuth tokens associated with the user.
-                retire_dot_oauth2_models(user)
-                AccountRecovery.retire_recovery_email(user.id)
-        except KeyError:
-            error_message = f'Username not specified {user}'
-            logger.error(error_message)
-            raise CommandError(error_message)  # lint-amnesty, pylint: disable=raise-missing-from
-        except user_model.DoesNotExist:
-            error_message = f'The user "{user.username}" does not exist.'
-            logger.error(error_message)
-            raise CommandError(error_message)  # lint-amnesty, pylint: disable=raise-missing-from
-        except Exception as exc:  # pylint: disable=broad-except
-            error_message = f'500 error deactivating account {exc}'
+            userfile = options['user_file']
+            userinfo = open(userfile, 'r')
+        except Exception as exc:
+            error_message = f'Error while reading file: {exc}'
             logger.error(error_message)
             raise CommandError(error_message)  # lint-amnesty, pylint: disable=raise-missing-from
 
-        logger.info("User succesfully moved to the retirment pipeline")
+        for record in userinfo:
```

Comment:
Are we worried about partial success while iterating through this file? If for example the process was killed when 50 users into a 100 user file what would happen if the file was re-run?

Reply:
@dnuzz , it think its a valid concern I missed it sorry. so I think what we want to do is that maybe let say if we have an exception raised for a user, we might want to log exception for that user and keep processing for other users, does that sounds fine to you?. I think in this way we might need to re-process the even if it failed for a user. If you have suggestions I would like to hear more from you.

Reply:
So this depends on a few things from my perspective: Is it possible a user is re-run through the queue if we don't successfully process them? And how often do failures occur when trying to retire users? 

Normally I would take the approach of trying to retire all the users in the file in a transaction together, or failing and reporting the user and file that failed so that someone can investigate. In my opinion this makes more sense if when a user is polled from the LMS api they are removed from that queue (message deleted) but your approach above makes more sense if the user is placed back into the queue (message churned/ re-queued)

Another question I have is if there is a way to access files post-failure? I assume they exist on the Jenkins instance, I am not sure what the clean-up procedure is there though or if they could be exported as an artifact.

Reply:
So regarding how often failure occur. As per my knowledge I have seen any error or complaints that when trying to add user to the retirement queue that process failed.
The next thing is that if there is any error while adding user to the retirement queue, the user won't be added to the retirement queue, so I think to answer the question regarding re-processing user from the retirement queue, since the user won't get added to queue in case of failure, we won't reprocessing the user from the retirement queue I think we will be re-processing user from the file itself.
Regarding how to access the file that process is also unclear to me as well, as I have been trying to get answers myself how this command will be used in upstream processes.